synology_dsm/synology_dsm.py

```diff
@@ -2,6 +2,7 @@
 """Class to interact with Synology DSM."""
 import socket
 import urllib3
+from future.moves.urllib.parse import quote

```

Comment:
```python
import six
if six.PY2:
    from future.moves.urllib.parse import urlencode
else:
    from urllib.parse import urlencode  # pylint: disable=import-error,no-name-in-module
```

As in tests

Reply:
So the problem is, we need to use `quote_via=quote` in `urlencode` in order to encode space into `%20`. The `quote_via` option is only available in Python 3.5+. So for Python 3.4, we still need to import from the `future` package. Not sure if there is a better workaround.

Reply:
I'll setup Python2 and 3.4 environment and do some local tests.

Reply:
So, I did some test and seems `quote_via` is not available in both Python 3.4 and Python 2 even import from the `future` package. Without that option, `urlencode` will still encode space into `+`. So I guess in order to support those versions, we still need to use the `quote` one.